packages/components/src/shared/src/useSafeLayoutEffect.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from "react";
+import { isBrowser } from "./dom";
+
+/**
+ * useSafeLayoutEffect enables us to safely call `useLayoutEffect` on the browser
+ * (for SSR reasons)
+ *
+ * React currently throws a warning when using useLayoutEffect on the server.
+ * To get around it, we can conditionally useEffect on the server (no-op) and
+ * useLayoutEffect in the browser.
+ *
+ * @see https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85
+ */
+export const useSafeLayoutEffect = isBrowser
+    ? React.useLayoutEffect
+    : React.useEffect;
```

Comment:
Euh!!! Not sure about this.... useEffect is very different than useLayoutEffect. One is synchronous and one is asychronous. Event when 100% on the client they are not interchangeable.

Reply:
This is a commonly use pattern: 
Chakra: https://github.dev/chakra-ui/chakra-ui/blob/8e201b9f84a914e58cdc9ee90b3e03269fabf4d7/packages/hooks/src/use-safe-layout-effect.ts#L12
Spectrum has something similar: https://github.dev/adobe/react-spectrum/blob/91ca94fe52840b7a32b961ec08208f5fbdf65697/packages/%40react-aria/utils/src/useLayoutEffect.ts#L21
Dan Abramov suggests it : https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85

the idea is that
"If you use server rendering, keep in mind that neither useLayoutEffect nor useEffect can run until the JavaScript is downloaded."